lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/tasklet/RunningExecutors.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver.tasklet;
+
+import org.apache.reef.vortex.driver.scheduler.TaskletScheduler;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.*;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Keeps track of all running TaskletExecutors and Tasklets.
+ * todo: create unit tests to test possible race conditions
+ */
+public class RunningExecutors {
```

Comment:
We're missing profiling here and there (e.g., number of pending tasklets over time, etc.). Let's create a JIRA issue for that.